test/OpenTelemetry.Instrumentation.Http.Tests/HttpClientTests.Basic.cs

```diff
@@ -584,6 +590,105 @@ public async Task HttpClientInstrumentationDoesNotReportExceptionEventOnErrorRes
             Assert.Empty(exportedItems[0].Events);
         }
 
+        [Theory]
+        [InlineData(true, true)]
+        [InlineData(true, false)]
+        [InlineData(false, true)]
+        [InlineData(false, false)]
+        public async Task CustomPropagatorCalled(bool sample, bool createParentActivity)
+        {
+            ActivityContext parentContext = default;
+            ActivityContext contextFromPropagator = default;
+
+            var propagator = new Mock<TextMapPropagator>();
+            propagator.Setup(m => m.Inject(It.IsAny<PropagationContext>(), It.IsAny<HttpRequestMessage>(), It.IsAny<Action<HttpRequestMessage, string, string>>()))
+                .Callback<PropagationContext, HttpRequestMessage, Action<HttpRequestMessage, string, string>>((context, message, action) =>
+                {
+                    contextFromPropagator = context.ActivityContext;
+
+                    string traceparent = $"00/{context.ActivityContext.TraceId}/{context.ActivityContext.SpanId}/01";
+
+                    message.RequestUri = new Uri(message.RequestUri.OriginalString + $"?expectedTraceparent={traceparent}");
+
+                    action(message, "traceparent", traceparent);
+                    action(message, "tracestate", Activity.Current.TraceStateString);
+                });
+
+            Sdk.SetDefaultTextMapPropagator(propagator.Object);
+
+            var exportedItems = new List<Activity>();
+
+            using (var traceprovider = Sdk.CreateTracerProviderBuilder()
+               .AddHttpClientInstrumentation()
+               .AddInMemoryExporter(exportedItems)
+               .SetSampler(sample ? new ParentBasedSampler(new AlwaysOnSampler()) : new AlwaysOffSampler())
+               .Build())
+            {
+                Activity parent = null;
+                if (createParentActivity)
+                {
+                    parent = new Activity("parent")
+                        .SetIdFormat(ActivityIdFormat.W3C)
+                        .Start();
+
+                    parent.TraceStateString = "k1=v1,k2=v2";
+                    parent.ActivityTraceFlags = ActivityTraceFlags.Recorded;
+
+                    parentContext = parent.Context;
+                }
+
+                var request = new HttpRequestMessage
+                {
+                    RequestUri = new Uri(this.url),
+                    Method = new HttpMethod("GET"),
+                };
+
+                using var c = new HttpClient();
+                await c.SendAsync(request);
+
+                parent?.Stop();
+            }
+
+            if (!sample)
+            {
+                Assert.Empty(exportedItems);
+            }
+            else
+            {
+                Assert.Single(exportedItems);
```

Comment:
For test case `sample == true && createParentActivity == true` shouldn't we expect nothing exported?

That is, the parent activity that is created is not sampled, right? So per the default `ParentBasedSampler` `LocalParentNotSampled` sampler defaults to `AlwaysOff`.

Reply:
> That is, the parent activity that is created is not sampled, right?

It is fake sampled 🤣 

```csharp
   parent = new Activity("parent")
      .SetIdFormat(ActivityIdFormat.W3C)
      .Start();

   parent.TraceStateString = "k1=v1,k2=v2";
   parent.ActivityTraceFlags = ActivityTraceFlags.Recorded; // <- Here
```